pilot/pkg/model/service.go

```diff
@@ -417,6 +417,7 @@ type ServiceDiscovery interface {
 	// determine the intended destination of a connection without a Host header on the request.
 	GetProxyServiceInstances(*Proxy) ([]*ServiceInstance, error)
 
+	// GetProxyWorkloadLabels returns the labels of the given proxy.
```

Comment:
I don't remember - what is the use case for this method ? For the connected envoy we now get them from metadata.
For EDS - the incremental EDS and associated structures get them from registry. Why do we need to make additional calls to the registry, would getting them from the same structs where EDS stores data be simpler  ?

Reply:
For k8s, it is possible using the structure in DiscoveryServer

But for other registries, what if the labels changed. This is to update them.

Reply:
We removed the workload labels